engine/src/main/java/org/terasology/rendering/dag/FinalPostProcessingNode.java

```diff
@@ -0,0 +1,226 @@
+     * adds depth-of-field blur, motion blur and film grain to the rendering
+     * obtained so far. If OculusVR support is enabled, it composes (over two
+     * calls) the images for each eye into a single image, and applies a distortion
+     * pattern to each, to match the optics in the OculusVR headset.
+     * <p>
+     * Finally, it either sends the image to the display or, when taking a screenshot,
+     * instructs the FrameBuffersManager to save it to a file.
+     * <p>
+     * worldRenderer.getCurrentRenderStage() Can be MONO, LEFT_EYE or RIGHT_EYE, and communicates to the method weather
+     * it is dealing with a standard display or an OculusVR setup, and in the
+     * latter case, which eye is currently being rendered. Notice that if the
+     * OculusVR support is enabled, the image is sent to screen or saved to
+     * file only when the value passed in is RIGHT_EYE, as the processing for
+     * the LEFT_EYE comes first and leads to an incomplete image.
+     */
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/finalpostprocessing");
+
+        ocUndistorted = frameBuffersManager.getFBO("ocUndistorted");
+        sceneFinal = frameBuffersManager.getFBO("sceneFinal");
+        sceneOpaque = frameBuffersManager.getFBO("sceneOpaque");
+
+        fullScale = sceneOpaque.dimensions();
+
+        if (!renderingDebugConfig.isEnabled()) {
+            finalPost.enable();
+        } else {
+            debug.enable();
+        }
+
+        if (!renderingConfig.isOculusVrSupport()) {
+            renderFinalMonoImage();
+        } else {
+            renderFinalStereoImage(worldRenderer.getCurrentRenderStage());
+        }
+
+        PerformanceMonitor.endActivity();
+    }
+
+    private void renderFinalMonoImage() {
+        if (postProcessor.isNotTakingScreenshot()) {
+            bindDisplay();
+            renderFullscreenQuad(0, 0, Display.getWidth(), Display.getHeight());
+
+        } else {
+            sceneFinal.bind();
+
+            glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+            renderFullscreenQuad(0, 0, fullScale.width(), fullScale.height());
+
+            postProcessor.saveScreenshot();
+            // when saving a screenshot we do not send the image to screen,
+            // to avoid the brief one-frame flicker of the screenshot
+
+            // This is needed to avoid the UI (which is not currently saved within the
+            // screenshot) being rendered for one frame with buffers.sceneFinal size.
+            setViewportToSizeOf(sceneOpaque);
+        }
+    }
+
+    // TODO: have a flag to invert the eyes (Cross Eye 3D), as mentioned in
+    // TODO: http://forum.terasology.org/threads/happy-coding.1018/#post-11264
+    private void renderFinalStereoImage(RenderingStage renderingStage) {
+        if (postProcessor.isNotTakingScreenshot()) {
+            sceneFinal.bind();
+        } else {
+            ocUndistorted.bind();
+        }
```

Comment:
This IF block only confuses things here as these bindings are used only if we are dealing with the left eye: when the right eye is being rendered the display or sceneFinal are bound respectively, overriding what the if blocks does. 

I'd say we move the IF block to within the LEFT_EYE case. Let's also verify it all still works though... =)